profiling/src/profile/mod.rs

```diff
@@ -260,9 +230,20 @@ impl UpscalingRule {
     }
 }
 
+// For testing and debugging purposes
+impl Profile {
+    pub fn num_aggregated_samples(&self) -> usize {
+        self.aggregated_samples.len()
+    }
+    pub fn num_timestamped_samples(&self) -> usize {
+        self.timestamped_samples.len()
+    }
```

Comment:
Minor: Should we prefix these methods with `testing_` or even `only_for_testing_`? The objective being to make it kinda ugly and obvious even when doing an IDE autocomplete that you shouldn't be using these methods :)

Reply:
done